src/codec/MapEntrySetCodec.ts

```diff
@@ -0,0 +1,49 @@
+/* tslint:disable */
+import ClientMessage = require('../ClientMessage');
+import {BitsUtil} from '../BitsUtil';
+import Address = require('../Address');
+import {Data} from '../serialization/Data';
+import {MapMessageType} from './MapMessageType';
+
+var REQUEST_TYPE = MapMessageType.MAP_ENTRYSET;
+var RESPONSE_TYPE = 117;
+var RETRYABLE = false;
+
+
+export class MapEntrySetCodec {
+
+
+    static calculateSize(name:string) {
+        // Calculates the request payload size
+        var dataSize:number = 0;
+        dataSize += BitsUtil.calculateSizeString(name);
+        return dataSize;
+    }
+
+    static encodeRequest(name:string) {
+        // Encode request into clientMessage
+        var clientMessage = ClientMessage.newClientMessage(this.calculateSize(name));
+        clientMessage.setMessageType(REQUEST_TYPE);
+        clientMessage.setRetryable(RETRYABLE);
+        clientMessage.appendString(name);
+        clientMessage.updateFrameLength();
+        return clientMessage;
+    }
+
+    static decodeResponse(clientMessage:ClientMessage, toObjectFunction:(data:Data) => any) {
+        // Decode response from client message
+        var parameters:any = {};
+        var responseSize = clientMessage.readInt32();
+        var response:any = [];
+        for (var responseIndex = 0; responseIndex <= responseSize; responseIndex++) {
+            var responseItem = clientMessage.readMapEntry();
+
+            response.push(responseItem)
+        }
+        parameters['response'] = response;
```

Comment:
you need to deserialize the result here. You need something like a custom List/array implementation that deserialize the item if accessed so you won't deserialize everything